### PR TITLE
Color libraries

### PR DESCRIPTION
**Pull request by Cenadros**

### CODE REVIEW

File: ui-src/parser/creators/createComponentsLibrary.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const createComponentsLibrary = async (file: PenpotFile) => {
   });
 
   for (const uiComponent of components) {
-    createComponentLibrary(file, uiComponent);
+    await createComponentLibrary(file, uiComponent);
```
Comment (jordisala1991) [inline ui-src/parser/creators/createComponentsLibrary.ts:26]: this should not be await

File: ui-src/parser/creators/createColorsLibrary.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    data: libraries.length
+  });
+
+  sendMessage({
+    type: 'PROGRESS_STEP',
+    data: 'libraries'
+  });
+
+  for (const library of libraries) {
+    for (let i = 0; i < library.fills.length; i++) {
```
Comment (jordisala1991) [inline ui-src/parser/creators/createColorsLibrary.ts:22]: avoid using 1 length variables, maybe is better to call it index?

File: ui-src/parser/creators/createComponentsLibrary.ts (hunk 1 above)
Comment (Cenadros) [reply ui-src/parser/creators/createComponentsLibrary.ts]: Fixed

File: ui-src/parser/creators/createColorsLibrary.ts (hunk 2 above)
Comment (Cenadros) [reply ui-src/parser/creators/createColorsLibrary.ts]: Fixed

File: ui-src/parser/parse.ts [hunk 3]
```diff
@@ -41,6 +44,43 @@ const optimizeImages = async (images: Record<string, Uint8Array>) => {
...
+    data: stylesToRegister.length
+  });
+
+  sendMessage({
+    type: 'PROGRESS_STEP',
+    data: 'format'
+  });
+
+  for (const [key, fillStyle] of stylesToRegister) {
+    for (let i = 0; i < fillStyle.fills.length; i++) {
```
[the remark this answers is not in the document]
Comment (Cenadros) [reply ui-src/parser/parse.ts]: Fixed

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**